### PR TITLE
Move e2e test steps from workflows into composite action 

### PR DESCRIPTION
**Pull request by wojteknowacki**
1. Moved e2e steps into github actions:
- on each pr
- scheduled
- triggered manually
DID not moved release step since more logic need to be changed
2. Changed scheduled tests to be trigged on Monday, Wednesday, Friday
3. Moved reports artifacts steps into action
4. Moved testmo (reporter) steps into actions
5. Bumped checkout action  v3 -> v4 in workflows
6. Added testmo into e2e nightly tests
7. Changed ubuntu-latest -> ubuntu-22.04 in nightly workflow

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
9. [ ] Attributes `data-test-id` are added for new elements
10. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
11. [ ] Your code works with the latest stable version of the core
12. [ ] I added changesets and [read good practices](../.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
13. [ ] giftCard
14. [ ] homePage
15. [ ] login
16. [ ] menuNavigation
17. [ ] navigation
18. [ ] orders
19. [ ] pages
20. [ ] payments
21. [ ] permissions
22. [ ] plugins
23. [ ] productType
24. [ ] products
25. [ ] sales
26. [ ] shipping
27. [ ] translations
28. [ ] variants
29. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

**Review by yellowee — DISMISSED** [review]

**Review by poulch — DISMISSED** [review]

File: .github/actions/e2e/action.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    required: true
+  secondUserName:
+    description: "Name of the second user which e2e uses"
+    required: true
+  userPassword:
+    description: "User password for e2e"
+    required: true
+  permissionsUserPassword:
+    description: "User with limited permissions password"
+    required: true
```
Comment (karola312) [inline .github/actions/e2e/action.yml:42]: I think we are always using same user, password etc. Do we need to have it in inputs? I think it always will be secrets.CYPRESS_USER_NAME, etc

File: .github/actions/e2e/action.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  githubToken:
+    description: "GH token"
+    required: false
```
Comment (karola312) [inline .github/actions/e2e/action.yml:45]: Where we are using value from this input?

Comment (wojteknowacki) [reply .github/actions/e2e/action.yml:45]: .github/workflows/tests-nightly.yml 180 line

Comment (wojteknowacki) [reply .github/actions/e2e/action.yml:45]: I did not validate inputs in workflows, I just refactored what we had - might be it is not needed

File: .github/actions/e2e/action.yml (hunk 1 above)
Comment (wojteknowacki) [reply .github/actions/e2e/action.yml:42]: Yeah, you are right - I will make sure to fix it when new introducing PW in place of cypress. Is that ok?

**Review by szczecha — APPROVED** [review]

**Review by michalina-graczyk — DISMISSED** [review]

File: .github/actions/e2e/action.yml (hunk 2 above)
Comment (NyanKiyoshi) [reply .github/actions/e2e/action.yml:45]: @wojteknowacki I don't see the token being used in this file (`.github/actions/e2e/action.yml`), probably a leftover?

Also it seems like we don't need to pass it, you can retrieve it from `${{ github.token }}` inside actions (https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)

File: .github/workflows/tests-nightly.yml [hunk 3]
```diff
@@ -453,14 +410,14 @@ jobs:
...
-          node cypressTestsHelpers/getFailedTests.js \
-          --dashboard_url "$dashboard_url"
+          - name: Get tests results
+          id: get-tests-results
+          env:
+            tests_status: ${{ needs.run-tests-in-parallel.outputs.status }}
+            dashboard_url: ${{ needs.run-tests-in-parallel.outputs.dashboard_url }}
+          run: |
+            node cypressTestsHelpers/getFailedTests.js \
+            --dashboard_url "$dashboard_url"
```
Comment (NyanKiyoshi) [inline .github/workflows/tests-nightly.yml:420]: YAML syntax is invalid here

File: .github/workflows/tests-nightly.yml [hunk 4]
```diff
@@ -471,4 +428,19 @@ jobs:
           JOB_TITLE: "A lot of tests failed, something is probably broken - ${{ needs.run-tests-in-parallel.outputs.dashboard_url }}"
           JOB_KIND: "Tests"
         run: |
-          python3 .github/workflows/notify/notify-slack.py
+          python3 notify/notify-slack.py
```
Comment (NyanKiyoshi) [inline .github/workflows/tests-nightly.yml:431]: Doesn't seem valid as it appears that the current working directory is the root of the repo:

```suggestion
          python3 .github/workflows/notify/notify-slack.py
```

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**